sqlcapture/capture.go

```diff
@@ -291,15 +291,15 @@ func (c *Capture) streamToWatermark(replStream ReplicationStream, watermark stri
 		// While a table is being backfilled, events occurring *before* the current scan point
 		// will be emitted, while events *after* that point will be patched (or ignored) into
 		// the buffered resultSet.
-		var rowKey, err = encodeRowKey(tableState.KeyColumns, event.KeyFields())
+		var rowKey, err = encodeRowKey(tableState.KeyColumns, event.KeyFields(), c.Database)
 		if err != nil {
 			return fmt.Errorf("error encoding row key: %w", err)
 		}
 		if compareTuples(rowKey, tableState.Scanned) <= 0 {
 			if err := c.handleChangeEvent(event); err != nil {
 				return fmt.Errorf("error handling replication event: %w", err)
 			}
-		} else if err := results.Patch(streamID, event); err != nil {
+		} else if err := results.Patch(streamID, event, c.Database); err != nil {
```

Comment:
This is an entirely adequate way of doing things, but since we're adding an argument to the `Patch` method anyway it would be slightly more efficient to pass in `rowKey` which we already computed a few lines prior.